bootstore/src/schemes/v0/fsm.rs

```diff
@@ -0,0 +1,479 @@
+
+    pub fn common_data(&self) -> &FsmCommonData {
+        &self.common
+    }
+
+    /// This call is triggered locally as a result of RSS running
+    /// It may only be called once, which is enforced by checking to see if
+    /// we already are in `State::Uninitialized`.
+    pub fn init_rack(
+        &mut self,
+        rack_uuid: Uuid,
+        initial_membership: BTreeSet<Baseboard>,
+    ) -> Output {
+        let State::Uninitialized(_) = self.state.as_ref().unwrap() else {
+            return ApiError::RackAlreadyInitialized.into();
+        };
+        let total_members = initial_membership.len();
+        match create_pkgs(rack_uuid, initial_membership.clone()) {
+            Ok(pkgs) => {
+                // We learn our pkg by iterating
+                let mut our_pkg = None;
+                let request_id = Uuid::new_v4();
+                let envelopes: Vec<_> = pkgs
+                    .expose_secret()
+                    .into_iter()
+                    .zip(initial_membership)
+                    .filter_map(|(pkg, peer)| {
+                        if peer == self.common.id {
+                            // Don't send a message to ourself
+                            our_pkg = Some(pkg.clone());
+                            None
+                        } else {
+                            Some(Envelope {
+                                to: peer,
+                                msg: Request {
+                                    id: request_id,
+                                    type_: RequestType::Init(pkg.clone()),
+                                }
+                                .into(),
+                            })
+                        }
+                    })
+                    .collect();
+                self.state = Some(State::InitialMember(InitialMemberState {
+                    pkg: our_pkg.unwrap(),
+                    distributed_shares: BTreeMap::new(),
+                    rack_init_state: Some(RackInitState::Running {
+                        start: self.common.clock,
+                        total_members,
+                        acks: BTreeSet::from([self.common.id.clone()]),
+                        unacked_envelopes: envelopes.clone(),
+                    }),
+                    pending_learn_requests: BTreeMap::new(),
+                }));
+                // Note that while we could only send the envelopes to
+                // connected peers, the network layer will just drop them.
+                // We already have a copy to send on connection stored in
+                // `Rack::InitState::Running.unacked_envelopes`.
+                Output { persist: true, envelopes, api_output: None }
+            }
+            Err(e) => return ApiError::RackInitFailed(e).into(),
+        }
+    }
+
+    /// Initialize a node added after rack initialization
+    pub fn init_learner(&mut self) -> Output {
+        let State::Uninitialized(_) = self.state.as_ref().unwrap() else {
```

Comment:
Since you don't actually use the captured value, how about
```suggestion
        if !matches!(self.state, Some(State::Uninitialized(..))) {
```

Reply:
No longer relevant as of [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)